core/src/main/java/org/apache/calcite/rex/RexFieldCollation.java

```diff
@@ -44,7 +45,7 @@ public class RexFieldCollation extends Pair<RexNode, ImmutableSet<SqlKind>> {
           .build();
 
   public RexFieldCollation(RexNode left, Set<SqlKind> right) {
-    super(left, KINDS.get(right));
+    super(left, Objects.requireNonNull(KINDS.get(right)));
```

Comment:
I tend to use a static import for Objects.requireNinNull to keep the code short.